### PR TITLE
tests: exclude sam0 boards from tests that will not build on them

### PR DESCRIPTION
**Pull request by galak**
the async uart and spi loopback tests will not build on SAM0 based boards as they require dma to be configured and none of the boards do that.  So we exclude these boards from being built for these tests.

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]
LGTM

**Review by stephanosio — COMMENTED** [review]
As far as I can remember, these used to work. I wonder what changed ...

**Review by carlescufi — APPROVED** [review]

File: tests/drivers/uart/uart_async_api/testcase.yaml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+common:
+  platform_exclude: seeeduino_xiao serpente arduino_nano_33_iot atsamr21_xpro
```
Comment (nashif) [inline tests/drivers/uart/uart_async_api/testcase.yaml:2]: i thought we were trying to move away from this. How did this test started failing all of the sudden?

Comment (galak) [reply tests/drivers/uart/uart_async_api/testcase.yaml:2]: @nashif the change to use DEVICE_DT_GET changed a runtime error into a compile error.  So these tests would build before but would never run correctly.

One issue is that if we enable UART ASYNC support in Kconfig, than its assumed ALL uart instances support that which may not be the case.

Comment (galak) [reply tests/drivers/uart/uart_async_api/testcase.yaml:2]: I'm open to a better solution.

**Pull request merged**